haystack/nodes/file_converter/parsr.py

```diff
@@ -138,18 +153,25 @@ def convert(
                             text += f"{current_paragraph}\n\n"
 
                     elif element["type"] == "table":
-                        table = self._convert_table_element(element, parsr_output["pages"], page_idx, elem_idx, meta)
+                        table = self._convert_table_element(
+                            element, parsr_output["pages"], page_idx, elem_idx, meta, id_hash_keys
+                        )
                         tables.append(table)
 
         if valid_languages:
-            file_text = text + " ".join([cell for table in tables for row in table["content"] for cell in row])
+            file_text = text
+            for table in tables:
+                assert isinstance(table.content, pd.DataFrame)
```

Comment:
Same as above

Reply:
Yes, the same holds for the other assert statements that were newly introduced. 👍